### PR TITLE
Set default coinType for PaySui txn sui

### PR DESCRIPTION
**Pull request by Jibz1**
- Set default coinType for PaySui txn SUI. 
- Show the amount for fail txn.  
- Applied the same to the wallet
- show amount for failed txns
- https://explorer-git-gj-explorer-fix-failed-txn-amount-1-mysten-labs.vercel.app/transaction/G4dWuiojLwuZ78bEtQUzwKu4B5JnrXD91AeyaMFLubYH
Screenshot 2023-02-17 at 2 24 22 PM

### CODE REVIEW

File: apps/explorer/src/utils/getAmount.ts [hunk 1]
```diff
@@ -72,27 +74,26 @@ export function getTransfersAmount(
             : null;
     }
 
-    const paySuiData =
-        getPaySuiTransaction(txnData) ?? getPayTransaction(txnData);
+    const payData = getPaySuiTransaction(txnData) ?? getPayTransaction(txnData);
 
-    const amountByRecipient = paySuiData?.recipients.reduce(
+    const amountByRecipient = payData?.recipients.reduce(
         (acc, value, index) => {
```
Comment (pchrysochoidis) [inline apps/explorer/src/utils/getAmount.ts:80]: value is the address of the recipient right? can we rename it to something like `recipientAddress`?

File: apps/explorer/src/utils/getAmount.ts [hunk 2]
```diff
@@ -72,27 +74,26 @@ export function getTransfersAmount(
...
-                    address:
-                        paySuiData.recipients[index] ||
-                        paySuiData.recipients[0],
+                    // for PaySuiTransaction the coinType is SUI
+                    coinType:
+                        txKindName === 'PaySui'
+                            ? SUI_TYPE_ARG
+                            : getCoinType(
+                                  txnEffect || null,
+                                  payData.recipients[index] ||
```
Comment (pchrysochoidis) [inline apps/explorer/src/utils/getAmount.ts:93]: why not just `value` (or `recipientAddress` if you rename it)

Comment (pchrysochoidis) [reply apps/explorer/src/utils/getAmount.ts:93]: and I think `payData.recipients[0]` is unnecessary

File: apps/explorer/src/utils/getAmount.ts [hunk 3]
```diff
@@ -72,27 +74,26 @@ export function getTransfersAmount(
...
+                                      payData.recipients[0]
+                              ),
+                    address: payData.recipients[index] || payData.recipients[0],
```
Comment (pchrysochoidis) [inline apps/explorer/src/utils/getAmount.ts:96]:
```suggestion
                    address: recipientAddress,
```

Comment (Jibz1) [reply apps/explorer/src/utils/getAmount.ts]: good catch. Done

File: apps/explorer/src/utils/getAmount.ts (hunk 2 above)
Comment (Jibz1) [reply apps/explorer/src/utils/getAmount.ts]: done

File: apps/explorer/src/utils/getAmount.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (Jibz1) [reply apps/explorer/src/utils/getAmount.ts]: done

**Review by pchrysochoidis — APPROVED** [review]

**Pull request merged**